File: book-social-network-ui/src/app/models/book.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  description: string;

  available: boolean;
  archived: boolean;

  ownerId?: number | null;
  borrowedById?: number | null;

  ownerName?: string | null;
  borrowedByName?: string | null;
}

File: booksocialnetwork/src/main/java/com/booknetwork/booksocialnetwork/entity/Book.java
package com.booknetwork.booksocialnetwork.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String author;
    private String description;

    private boolean archived = false;
    private boolean available = true;

    @ManyToOne
    @JsonIgnore
    private User owner;

    @ManyToOne
    @JsonIgnore
    private User borrowedBy;
}
